dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -98,6 +99,14 @@ public abstract class KernelFunction
     /// </remarks>
     public IReadOnlyDictionary<string, PromptExecutionSettings>? ExecutionSettings { get; }
 
+    /// <summary>
+    /// Gets the underlying <see cref="MethodInfo"/> that this function might be wrapping.
+    /// </summary>
+    /// <remarks>
+    /// Provides additional metadata on the function and its signature. Implementations not wrapping .NET methods may return null.
+    /// </remarks>
+    public MethodInfo? UnderlyingMethod { get; internal init; }
```

Comment:
We already use `Inner` for breaking glass scenarios in our contents, wonder if that's better naming here.

Suggest adding experimental tag as well.

```suggestion
    public MethodInfo? InnerMethod { get; internal init; }
```